### PR TITLE
libvpx: update to version 1.8.2

### PR DESCRIPTION
**Pull request by BKPepe**
Maintainer: @luizluca 
Compile tested: Turris 1.1, powerpc_8540, OpenWrt master

Description:
- Update to version [1.8.2](https://chromium.googlesource.com/webm/libvpx/+/refs/tags/v1.8.2)
- Change URL to use HTTPS instead of HTTP
- Remove empty row between maintainer and license

### CODE REVIEW

File: libs/libvpx/Makefile [hunk 1]
```diff
@@ -9,20 +9,19 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=libvpx
-PKG_VERSION:=1.8.1
+PKG_VERSION:=1.8.2
 PKG_RELEASE:=1
 
 PKG_REV:=v$(PKG_VERSION)
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION)-$(PKG_REV).tar.gz
```
Comment (neheb) [inline libs/libvpx/Makefile:16]: Should be removed.

File: libs/libvpx/Makefile [hunk 2]
```diff
@@ -9,20 +9,19 @@
...
-PKG_MIRROR_HASH:=6703842a870727b621a82efe5b2ff6f5553d41f7b0905dd4fde1f8bdf062d6ea
+PKG_MIRROR_HASH:=51e871a928fe98f14fd08285cb9b64c0d540b36b630ee7d47bc464e909366db7
 PKG_SOURCE_URL:=https://chromium.googlesource.com/webm/libvpx
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_VERSION:=$(PKG_REV)
```
Comment (neheb) [inline libs/libvpx/Makefile:20]: PKG_REV should be replaced and v$(PKG_VERSION) placed here.

File: libs/libvpx/Makefile [hunk 3]
```diff
@@ -9,20 +9,19 @@
...
 PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
```
Comment (neheb) [inline libs/libvpx/Makefile:21]: Should be removed.

File: libs/libvpx/Makefile [hunk 4]
```diff
@@ -9,20 +9,19 @@
...
 PKG_CPE_ID:=cpe:/a:john_koleszar:libvpx
 
 PKG_MAINTAINER:=Luiz Angelo Daros de Luca <luizluca@gmail.com>
-
 PKG_LICENSE:=BSD-3-Clause
 PKG_LICENSE_FILES:=LICENSE
```
Comment (neheb) [inline libs/libvpx/Makefile:27]: PKG_BUILD_PARALLEL should be placed later on.

**Comment by BKPepe** [comment]
Yeah, I agree with the suggested comments. I saw it and I was thinking if I should do just simple bump and do Makefile polishing now or later.

File: libs/libvpx/Makefile [hunk 5]
```diff
@@ -12,18 +12,17 @@ PKG_NAME:=libvpx
...
-PKG_MIRROR_HASH:=51e871a928fe98f14fd08285cb9b64c0d540b36b630ee7d47bc464e909366db7
-PKG_SOURCE_URL:=https://chromium.googlesource.com/webm/libvpx
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_VERSION:=$(PKG_REV)
-PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
-PKG_CPE_ID:=cpe:/a:john_koleszar:libvpx
+PKG_SOURCE_URL:=https://chromium.googlesource.com/webm/libvpx
+PKG_MIRROR_HASH:=51e871a928fe98f14fd08285cb9b64c0d540b36b630ee7d47bc464e909366db7
+PKG_SOURCE_VERSION:=v$(PKG_VERSION)
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_SOURCE_VERSION).tar.gz
```
Comment (neheb) [inline libs/libvpx/Makefile:19]: Why is this here? The default uses .xz archives.

**Comment by luizluca** [comment]
Looks good, thanks. I did more compiling testing on different targets with no errors.

However, I failed to to a runtime test as gstreamer seems to be broken (tested both with x86_64 and malta):

```
root@OpenWrt:~# gst-inspect-1.0 | cat
staticelements:  bin: Generic bin
staticelements:  pipeline: Pipeline object

Total count: 1 plugin, 2 features
root@OpenWrt:~# gst-typefind-1.0  | cat
[ 1297.929708] gst-typefind-1.[5222]: segfault at 58 ip 00007f79917bd91f sp 00007ffd32df6890 error 4 in libglib-2.0.so.0.5800.3[7f7991775000+6b000]
[ 1297.932400] Code: 63 fe ff eb 10 48 8d 3d 40 89 07 00 45 31 e4 67 e8 2c 63 fe ff 4c 89 e7 67 e8 1e a3 ff ff 4c 8b 23 4d 85 e4 74 2a 49 8b 34 24 <48> 8b 46 58 48 85 c0 75 07 4d 8b 64 24 08 eb e7 48 8b 56 28 48 8b
Segmentation fault
```

For reference, I normally do this test:

```
root@OpenWrt:~# opkg update
root@OpenWrt:~# opkg install /tmp/libvpx*.ipk /tmp/gst1-mod-vpx*.ipk
root@OpenWrt:~# opkg install gstreamer1-utils gst1-mod-vpx gst1-mod-videotestsrc gst1-mod-matroska gst1-mod-hls uhttpd
root@OpenWrt:~# /etc/init.d/uhttpd start
root@OpenWrt:~# cd /www && gst-launch-1.0 videotestsrc is-live=true ! vp8enc deadline=1 cpu-used=4 lag-in-frames=0 ! webmmux streamable=true ! hlssink max-files=5 target-duration=5
root@OpenWrt:~# cd /www && gst-launch-1.0 videotestsrc is-live=true ! vp9enc deadline=1 cpu-used=4 lag-in-frames=0 ! webmmux streamable=true ! hlssink max-files=5 target-duration=5

PC ~$ gst-play-1.0 http://192.168.1.1/playlist.m3u8
```

File: libs/libvpx/Makefile (hunk 5 above)
Comment (luizluca) [reply libs/libvpx/Makefile:19]: Sorry @neheb , I missed your comment while I was doing local testing. It indeed needs to go. I saves about 1MB. However, I already merged this. I'll do another commit for this.
